### ISSUE TITLE
[BUG] StatsForecastAutoARIMA takes significantly longer to fit compared to AutoARIMA from nixtla

### ISSUE THREAD

**Issue by surajjamodkar** [sktime/sktime#6710]
**Describe the bug**

**To Reproduce**

```python
%%time 
from statsforecast.models import AutoARIMA
from statsforecast import StatsForecast
models = [AutoARIMA()]
sf = StatsForecast(df=main_df2_train.reset_index(),
                   models=models,
                   freq='M', 
                   n_jobs=-1)
sf.fit()

%%time 
from sktime.forecasting.statsforecast import StatsForecastAutoARIMA
forecaster = StatsForecastAutoARIMA(n_jobs = -1)
forecaster.fit(main_df2_train)

```

**Expected behavior**

**Additional context**

**Versions**
<details>

</details>

**Comment by fkiraly** [comment]
FYI @yarnabrina, is this forwarding the `n_jobs` parameter correctly?

**Comment by fkiraly** [comment]
I think I know what is going on: the wrapper (originally contributed by `nixtla` authors) forwards the time series only by instance, not the entire panel.

So the `nixtla` parallelization cannot be used, however you can use the `sktime` generic parallelization, via `set_config`, via the `backend:parallel` and `backend:parallel:params`, you can select any `joblib` compatible backend (incl `spark`) or `dask`, see the `set_config` documentation.

It would be interesting to see how this compares.

As an action, we should add a parameter imo that lets the user choose whether they want to use `nixtla` native parallelization, or `sktime` framework generic parallelization.

FYI @yarnabrina

**Comment by yarnabrina** [comment]
I think what @Xinyu-Wu-0000  is doing in #6666 should address this, if your per instance vs all together guess is right?

**Comment by fkiraly** [comment]
No, I think this is not a global forecast case but a panel forecast case, where all time series are being forecast in parallel, so #6666 would be unrelated.

What needs to be done imo is to write treatment for the case where the user passes `pd-multiindex` or `pd_multiindex_hier` typed data, to be handled internally with the `statsforecast` parallelization. I would also think a user would like to be able to switch the `nixtla` native parallelization on and off, because they may like to parallelize on another level of a more complex workflow, e.g., in `evaluate` or `ForecastingBenchmark` or the entire pipeline, etc.

**Comment by yarnabrina** [comment]
Not sure why I did not get the notifucation for this thread. Sorry for late response.

I understand I made a mistake that #6666 is for `neuralforecast` and not for `statsforecast`, but I am not sure I understand what you meant. Can you please confirm my understanding?

I understand that we need to not broadcast per panel/hierarchy in `sktime` side but let it happen at `statsforecast` side, which is also what Xinyu is doing. But here there is no need (and possibility) to have the ability to predict for different set of series, which is another (and main) requirement in #6666. Is this what you are saying?

I was exploring this, and `statsforecast` has a class `StatsForecast` which we were not using. We were instead calling the underlying model instances directly. We can try to switch to use via this class, and that will likely be enough, and probably quite similar type of changes in dataset formatting as in #6666. I am however not sure how do we measure improvement in speed as part of regular CI testing.
